server/routes/actions/actions_routes.py

```
@@ -0,0 +1,395 @@
+        return None, f"trigger_type must be one of {_VALID_TRIGGER_TYPES}"
+    return val, None
+
+
+def _validate_mode(body):
+    val = body["mode"]
+    if val not in _VALID_MODES:
+        return None, f"mode must be one of {_VALID_MODES}"
+    return val, None
+
+
+def _validate_enabled(body):
```

Ruff (0.15.12)

[warning] Missing return type annotation for private function `_validate_enabled`

(ANN202)